core/keyboard_nav/navigation.js

```diff
@@ -837,14 +837,19 @@ Blockly.navigation.onKeyPress = function(e) {
   var key = Blockly.user.keyMap.serializeKeyEvent(e);
   var action = Blockly.user.keyMap.getActionByKeyCode(key);
   var curNode = Blockly.navigation.cursor_.getCurNode();
+  var readOnly = Blockly.getMainWorkspace().options.readOnly;
   var actionHandled = false;
 
   if (action) {
-    if (curNode && curNode.getType() === Blockly.ASTNode.types.FIELD) {
-      actionHandled = curNode.getLocation().onBlocklyAction(action);
-    }
-    if (!actionHandled) {
+    if (readOnly && Blockly.navigation.READONLY_ACTION_LIST.indexOf(action) > -1) {
```

Comment:
If you swap these two it becomes
if (!readonly) {
  ...
} else if (Blockly.navigation.READONLY_ACTION_LIST.indexOf(action) > -1) {
  ...
}

Reply:
fixed